processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -638,16 +639,33 @@ private Assignment forgeMapping(SourceRHS sourceRHS) {
             if ( sourceType.isPrimitive() || targetType.isPrimitive() ) {
                 return null;
             }
+
             String name = getName( sourceType, targetType );
-            ForgedMethod forgedMethod = new ForgedMethod(
-                name,
-                sourceType,
-                targetType,
-                method.getMapperConfiguration(),
-                method.getExecutable(),
-                method.getContextParameters(),
-                getForgedMethodHistory( sourceRHS )
-            );
+            List<Parameter> parameters = new ArrayList( method.getContextParameters() );
+            ForgedMethod forgedMethod;
+            if ( method.isUpdateMethod() ) {
+                parameters.add( Parameter.forForgedMappingTarget( targetType ) );
```

Comment:
I think its more cleaner to return void. I struggled in the past with update methods that have both a return type and a mapping target. I still don't like that :).